### DISCUSSION TITLE
Adding favourites to AutoComplete

### DISCUSSION THREAD

**Discussion by joao-carneiro-twistag** [heroui-inc/heroui#2503] [Feature requests]
I'm currently using the Autocomplete component from NextUI in my project, and I'm facing a challenge with implementing a feature related to a button within each dropdown item.

I'm seeking guidance on how to properly implement a favorite toggle button within each dropdown item while ensuring that it does not interfere with the main action triggered by selecting an item. Any suggestions, code examples, or insights on how to achieve this would be greatly appreciated.

**Comment by alphaxek** [comment]
Hi @jambc91,

are you looking for something like this?
![autcom](https://github.com/nextui-org/nextui/assets/116849110/59932cc0-0d1e-4dc9-acbe-f7e09ae4745c)

**Comment by joao-carneiro-twistag** [comment]
Hey! thanks for reply! Yes but as you see, whenever you click in that *, the modal is closing and the item is being selected and it shouldn't... What i would like to have is an independent button on each item, whenever it triggers, don't call the main function from the autoComplete onSelectChange.
The final outcome would be, a favourite symbol on that '*', which would toggle upon clicked, without selecting the entry from the list

**Comment by alphaxek** [comment]
Hi @jambc91, I believe you might be misusing `Autocomplete`, may be what you are looking for is the functionality provided by a `Listbox` instead.

Please refer below:
https://nextui.org/docs/components/listbox

![listbox](https://github.com/nextui-org/nextui/assets/116849110/4e4175d8-5a6b-4c21-ad0e-d976d5a1c327)

**Reply by joao-carneiro-twistag** [reply]
Thanks again for the reply @alphaxek .
Well i'm looking for something just like Autocomplete, a dropdown, which also has the auto-complete feature ( typing ).
But in addiction to that, i wanted that each item in the dropdown, could have some independent button ( for my case, a sort of "favourites" toggle.
I've tried to add manually a button in those items, but upon clicked, it would also trigger the main click
I'm still a junior so there's a big chance i'm messing things up, aha

**Comment by joao-carneiro-twistag** [comment]
The main issue i found is, if i add a button to the dropdown item, upon clicked, i cannot preventDefault or stopPropagation from the AutoComplete, it simply doesn't work

**Comment by DanielReyes03** [comment]
Any update of this?
